src/atomdb/AtomDBSingleton.cc

```diff
@@ -11,14 +11,23 @@ shared_ptr<AtomDB> AtomDBSingleton::atom_db = shared_ptr<AtomDB>{};
 // --------------------------------------------------------------------------------
 // Public methods
 
-void AtomDBSingleton::init() {
+void AtomDBSingleton::init(atomdb_api_types::ATOMDB_TYPE atomdb_type) {
     if (AtomDBSingleton::initialized) {
         Utils::error(
             "AtomDBSingleton already initialized. AtomDBSingleton::init() should be called only once.");
     } else {
         AtomDBCacheSingleton::init();
-        RedisMongoDB::initialize_statics();
-        AtomDBSingleton::atom_db = shared_ptr<AtomDB>(new RedisMongoDB());
+
+        if (atomdb_type == atomdb_api_types::ATOMDB_TYPE::MORK_MONGODB) {
+            MorkMongoDB::initialize_statics();
+            AtomDBSingleton::atom_db = shared_ptr<AtomDB>(new MorkMongoDB());
+        } else if (atomdb_type == atomdb_api_types::ATOMDB_TYPE::REDIS_MONGODB) {
+            RedisMongoDB::initialize_statics();
+            AtomDBSingleton::atom_db = shared_ptr<AtomDB>(new RedisMongoDB());
+        } else {
+            Utils::error("Unknown atomdb_type passed to AtomDBSingleton::init().");
```

Comment:
Error messages should have any available info to help debugging. In this case, you should provide the passed string which is supposed to be wrong.